lib/src/main/scala/spinal/lib/bus/regif/BusIfAdapter/AhbLite3BusInterface.scala

```diff
@@ -16,22 +16,35 @@ case class AhbLite3BusInterface(bus: AhbLite3, sizeMap: SizeMapping, regPre: Str
   readError.setAsReg() init False
   readData.setAsReg()  init 0
 
-  val askWrite    = bus.HSEL & bus.HTRANS(1) === True &  bus.HWRITE
-  val askRead     = bus.HSEL & bus.HTRANS(1) === True & !bus.HWRITE
-  val doWrite     = bus.HREADYOUT & RegNext(askWrite, False)
-  val doRead      = bus.HREADYOUT & askRead
+  val askWrite    = bus.HSEL & bus.HTRANS(1) &  bus.HWRITE
+  val askRead     = bus.HSEL & bus.HTRANS(1) & !bus.HWRITE
+  val doWrite     = bus.HREADY & RegNext(askWrite, False)
+  val doRead      = bus.HREADY & askRead
 
   val addressDelay = RegNextWhen(bus.HADDR, askRead | askWrite)
 
-  bus.HREADYOUT := True
   bus.HRESP     := readError
   bus.HRDATA    := readData
 
+  readError.clearWhen(readError)
+  val readError_2ndcycle = RegNext(readError) init False
+
+  when(readError_2ndcycle){
+    bus.HREADYOUT := True
+    bus.HRESP     := True
+  } elsewhen (readError){
+    bus.HREADYOUT := False
+    bus.HRESP     := True
+  } otherwise {
+    bus.HREADYOUT := True
+    bus.HRESP     := False
+  }
+
   def readAddress():UInt   = bus.HADDR
   def writeAddress(): UInt = addressDelay
 
-  def readHalt(): Unit  = bus.HREADYOUT === False
-  def writeHalt(): Unit = bus.HREADYOUT === False
+  def readHalt(): Unit  = bus.HREADY === False
+  def writeHalt(): Unit = bus.HREADY === False
```

Comment:
This is wrong because here it should be HREADYOUT not HREADY, and the action is to drive it. Here you just build a wire which is `!bus.HREADY` and forget it.

@jijingg Have you checked #920?

Reply:
In fact, these two functions are not used in Busif